bin/cylc-cat-state

```diff
@@ -99,18 +95,37 @@ def _get_state_lines(suite):
     dao.select_task_pool(
         lambda row_idx, row: _callback_task_pool(data, row))
 
-    lines = [
-        r"run mode : %(run_mode)s" % data,
-        r"time : %(time_str)s (%(time_since_epoch)s)" % data,
-        r"initial cycle : %(initial_point)s" % data,
-        r"final cycle : %(final_point)s" % data]
-    lines += pickle.dumps(data["broadcast_states"]).splitlines()
-    lines.append(r"Begin task states")
-    for item in data["task_pool"]:
-        lines.append(
+    return data
+
+
+def extract_json(state):
+    """Map task pool DB fields onto cat-state output fields."""
+    return {
+        TaskID.get(task['name'], task['cycle']): {
+            out_key: task[db_key] for out_key, db_key in (
+                ('name', 'name'),
+                ('point', 'cycle'),
+                ('label', 'cycle'),
+                ('state', 'status'),
+                ('spawned', 'spawned')
+            )
+        } for task in state['task_pool']
+    }
+
+
+def extract_lines(state):
+    """Extract textoral representation of the state."""
+    yield r"run mode : %(run_mode)s" % state
+    yield r"time : %(time_str)s (%(time_since_epoch)s)" % state
+    yield r"initial cycle : %(initial_point)s" % state
+    yield r"final cycle : %(final_point)s" % state
+    yield pickle.dumps(state["broadcast_states"]).splitlines()
```

Comment:
We should aim to remove usage of pickle. Time to break compat? A TODO here is also acceptable if we want to defer to follow-on PR.

Reply:
Easy enough to swap out `pickle` for `json`.